sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -322,7 +322,8 @@ case class StaticInvoke(
     propagateNull: Boolean = true,
     returnNullable: Boolean = true,
     isDeterministic: Boolean = true,
-    scalarFunction: Option[ScalarFunction[_]] = None) extends InvokeLike {
+    scalarFunction: Option[ScalarFunction[_]] = None,
+    override val nullIntolerant: Boolean = false) extends InvokeLike {
```

Comment:
This flag is the same as `propagateNull`?

BTW I kind of like the name `propagateNull` more as it's more straightforward. But I'm not a native English speaker and I could be wrong. cc @dtenedor @srielau

Reply:
I guess they are the same too. If you don't mind， I'd like too keep the PR in AS-IS state to minimize the potential influence of this refactoring work. Since nullIntolerant defaults to false，propagateNull defaults true，mapping them blindly might result in many unsure plan changes，I‘d like to revist and revise them in followups in the meanwhile I'm fixing the RuntimeReplaceables which use StaticInvoke and Invoke heavily？ WDYT？ @cloud-fan

Reply:
nullIntolerant and propagateNull currently have opposite defaults in where they are defined and the places they are called, I guess we shall leave them AS-IS and change them more carefully

Reply:
> OK, but at lease we shouldn't add a new parameter here. We can just add
> 
> ```
> override def nullIntolerant = propagateNull
> ```

I have tested this change in a separate push, the actions did show some test failures to me. I suggest we do that change separately.

Reply:
Does this mean `propagateNull` and `nullIntolerant` are not exactly the same? I'd like to figure this out now, instead of adding a new parameter to `StaticInvoke` (not binary compatible) and then remove it later.

Reply:
I didn't mean that, I mean propagateNull lacks of some optimizations applied to the NullIntolerant trait

Reply:
So it's not even related to `Invoke`?